src/utils.c

```diff
@@ -0,0 +1,47 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   utils.c                                            :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: linhnguy <linhnguy@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/07/26 12:36:24 by linhnguy          #+#    #+#             */
+/*   Updated: 2024/07/26 19:45:12 by linhnguy         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../include/cub3d.h"
+
+int	err(char *str)
+{
+	ft_printf(2, "%s\n", str);
+	return (1);
+}
+
+char	*strdup_no_newline(const char *s1)
+{
+	int		i;
+	char	*ptr;
+
+	i = 0;
+	ptr = malloc (ft_strlen(s1) + 1);
+	if (ptr == 0)
+		return (NULL);
+	while (s1[i])
+	{
+		if (s1[i] == '\n')
+			i++;
+		ptr[i] = s1[i];
+		i ++;
```

Comment:
This iterates past null

Maybe just check if last char of s1 is \n then malloc sizeof(char) * ft_strlen(s1), copy until second last char and replace the last with \0

If there is no \n at the end of s1 just ft_strdup it?